opsmngr/live_migration.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2021 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package opsmngr
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	atlas "go.mongodb.org/atlas/mongodbatlas"
```

Comment:
Same as the other question, I'd have expected this on an OM/CM cli, not on Atlas

Reply:
You see this import because I am reusing some of the structs from the atlas go client such as `*atlas.LinkToken` and `*atlas.LiveMigration`. I have recently added this [Cloud Migration endpoints](https://docs.atlas.mongodb.com/reference/api/live-migration/) to the atlas go client  https://github.com/mongodb/go-client-mongodb-atlas/pull/239

Reply:
So, one of the things we did for MMS is _not_ reuse the models between Atlas and OM. The reasoning was that, even though it's a duplication, it prevents us from making a breaking change on one side by mistake.

Eg. we remove/rename a member of the linkToken on Atlas, then all existing OM clients will start to fail because the field no longer exists.

I think we should use the same model here, extending the OM go client and using those types.

Reply:
Sure, we can have a quick chat on Monday about the path forward, I don't have a strong opinion here, each have their merits.

Reply:
reusing works here as clients are released together* so changes go together and is easier to control, unlike atlas and ops manager which tend to go different paths after big releases. That said @andreaangiolillo, you should not be reusing `atlas.LiveMigration`, it doesn't have the same meaning for these endpoints as it does for Atlas

Reply:
@colm-quinn Do you consider this thread resolved?

Reply:
Yup, I'm good, thanks @gssbzn for the reasoning difference with our go clients.